### ISSUE TITLE
support for the no-metadata parameter in /exp

### ISSUE THREAD

**Issue by jerrinot** [questdb/questdb#4629]
### To reproduce

According to [docs](https://questdb.io/docs/reference/api/rest/#overview) the REST interface accept `nm` parameter which says:
> Skips the metadata section of the response when set to true.

However, it has no effect for me. I can see the [parameter is set](https://github.com/questdb/questdb/blob/9683d03251aa6284bf8d2de0d6d60884a477d893/core/src/main/java/io/questdb/cutlass/http/processors/TextQueryProcessorState.java#L50), but it's never read. 

### QuestDB version:

8.0.0

### OS, in case of Docker specify Docker and the Host OS:

irrelevant

### File System, in case of Docker specify Host File System:

irrelevant

### Full Name:

Jaromir Hamala

### Affiliation:

QuestDB

### Have you followed Linux, MacOs kernel configuration steps to increase Maximum open files and Maximum virtual memory areas limit?

- [X] Yes, I have

**Comment by javier** [comment]
Is this for `nm` or for `count`? I came here to report this same issue with count, because it is ignored and ALWAYS shown, but I can see the `nm` is honoured in version 8

Try this (it will have metadata, so column names)

```
curl "https://demo.questdb.io/exec?limit=0&count=false&query=select%20count()%20from%20trades"
```

Then this one 
```
curl "https://demo.questdb.io/exec?nm=true&limit=0&count=false&query=select%20count()%20from%20trades"
```

In this case I have a minimal response, but count is still there!

**Comment by jerrinot** [comment]
@javier very interesting! this was meant for `nm`. maybe I misread something. will explore it and report back.

**Comment by jerrinot** [comment]
ok, I see. the `nm` parameters works for the `/exec` endpoint (=JSON), but not for the `/exp` endpoint (=CSV)

**Comment by jerrinot** [comment]
@javier: about the `count` parameter - it works, but its semantic is non-intuitive. 
I believe it only has a meaning when combined with the `limit` parameter. 

See this (the default):
`$ curl -G "https://demo.questdb.io/exec" --data-urlencode "query=SELECT * from trades where symbol like 'ETH%';" --data-urlencode "limit=1"  --data-urlencode "nm=true"`

```json
{"dataset":[["ETH-USD","sell",2615.54,4.4E-4,"2022-03-08T18:03:57.609765Z"]],"count":1}
```
The result says `"count": 1` since that's what we got. It's the size of returned data. 
Even if the query itself might return more rows we instructed QuestDB REST interface to cut it right after a first row. 

Now compare it with this ("count=true"):
`$ curl -G "https://demo.questdb.io/exec" --data-urlencode "query=SELECT * from trades where symbol like 'ETH%';" --data-urlencode "limit=1"  --data-urlencode "nm=true" --data-urlencode "count=true"`

```json
{"dataset":[["ETH-USD","sell",2615.54,4.4E-4,"2022-03-08T18:03:57.609765Z"]],"count":295353869}
```

The result now includes the row count as returned by the SQL, regardless of the `limit` query parameter. 

So it looks like the `count=true` changes the meaning of the `count` field in the result from "how many rows this response includes" to "how many rows a query could return". 

The question is whether the current behaviour is desirable and we should just clarify docs or whether the current behaviour is purely accidental. 


WDYT?
